framework/src/bundle/BundlePrivate.cpp

```diff
@@ -588,7 +588,20 @@ std::exception_ptr BundlePrivate::Start0()
   coreCtx->listeners.BundleChanged(BundleEvent(BundleEvent::BUNDLE_STARTING, thisBundle));
   
   Any bundleActivatorVal = thisBundle.GetProperty(Constants::BUNDLE_ACTIVATOR);
-  bool useActivator = bundleActivatorVal.Empty() ? false : any_cast<bool>(bundleActivatorVal);
+  bool useActivator = false;
+  
+  if(!bundleActivatorVal.Empty())
+  {
+    try
+    {
+      useActivator = any_cast<bool>(bundleActivatorVal);
+    }
+    catch (const BadAnyCastException& ex)
+    {
+      DIAG_LOG(*coreCtx->sink) << ex.what();
```

Comment:
I think it would be more useful to clients if this is a ```FrameworkEvent``` of type ```FRAMEWORK_WARNING``` instead so that clients can resolve potentially incorrect manifests.